gui/ros_gui/gui/configpanel/configpanel.py

```diff
@@ -0,0 +1,357 @@
+        self.get_logger().debug(f'Got value(s) {future.result().values} for parameter(s) {param_names}')
+        for idx, param_name in enumerate(param_names):
+            param_type = future.result().values[idx].type
+            match param_type:
+                case ParameterType.PARAMETER_BOOL:
+                    self.parameters[param_name] = future.result().values[idx].bool_value
+                case ParameterType.PARAMETER_INTEGER:
+                    self.parameters[param_name] = future.result().values[idx].integer_value
+                case ParameterType.PARAMETER_DOUBLE:
+                    self.parameters[param_name] = future.result().values[idx].double_value
+                case ParameterType.PARAMETER_STRING:
+                    self.parameters[param_name] = future.result().values[idx].string_value
+                case ParameterType.PARAMETER_BYTE_ARRAY:
+                    self.parameters[param_name] = future.result().values[idx].byte_array_value
+                case ParameterType.PARAMETER_BOOL_ARRAY:
+                    self.parameters[param_name] = future.result().values[idx].bool_array_value
+                case ParameterType.PARAMETER_INTEGER_ARRAY:
+                    self.parameters[param_name] = future.result().values[idx].integer_array_value
+                case ParameterType.PARAMETER_DOUBLE_ARRAY:
+                    self.parameters[param_name] = future.result().values[idx].double_array_value
+                case ParameterType.PARAMETER_STRING_ARRAY:
+                    self.parameters[param_name] = future.result().values[idx].string_array_value
+                case _:
+                    self.get_logger().info(f'Parameter {param_name} has an unsupported type {param_type}')
+
+    async def pick_file(self, node_name, param_name) -> None:
+        """ Opens a file picker and sets the selected file as the parameter value.
+        
+        Args:
+            node_name (str): Name of the node that the parameter belongs to.
+            param_name (str): Name of the parameter.
+        """
+        result = await local_file_picker('~/.astazero/ATOS', multiple=False, show_hidden_files=True)
+        ui.notify(f'You selected {result}')
+        if not result:
+            return
+        self.parameters[param_name] = str(result)
+        self.set_parameter(node_name, param_name, str(result))
+
+    async def pick_files(self, node_name, param_name) -> None:
+        """ Opens a file picker and sets the selected files as the parameter value.
+
+        Args:
+            node_name (str): Name of the node that the parameter belongs to.
+            param_name (str): Name of the parameter.
+        """
+        result = await local_file_picker('~/.astazero/ATOS', multiple=True, show_hidden_files=True)
+        ui.notify(f'You selected {result}')
+        if not result:
+            return
+        self.parameters[param_name] = str(result)
+        self.set_parameter(node_name, param_name, str(result))
```

Comment:
```suggestion
```

Reply:
duplicate?

Reply:
There is a very minor difference compared to "pick_file" which is that in "local_file_picker()" there is a flag "multiple" indicating if we can select more than one file or not.